shoes-core/lib/shoes/arc.rb

```diff
@@ -1,9 +1,24 @@
 # frozen_string_literal: true
 
 class Shoes
+  # Arc.  A basic element representing a curve of a circle or an oval.
+  #
+  # @example A simple arc which describes the bottom half of a circle and uses the centered style.
+  #   arc 200, 200, 100, 100, 0, Shoes::PI, center: true
+  # @example An arc which describes the top half of a circle.
+  #   arc 200, 200, 100, 100, Shoes::PI, 0
+  # @note Angle is the gradient angle used across all art elements. Angle1/2 are the angles of
+  #   the arc itself!
+  # @author Jason Clark
+  # @param left [Integer] The number of pixels from the left side of the window.
+  # @param top [Integer] The number of pixels from the top side of the window
+  # @param width [Integer] The width of the arc element.
+  # @param height [Integer] The height of the arc element.
+  # @param angle1 [Float] The first angle of the arc to the center point, in Radians,
+  #   starting from the 3 o'clock position.
+  # @param angle2 [Float] The second angle of the arc to the center point, in Radians.
+  # @param [Hash] Style hash
```

Comment:
I believe yard also supports describing the `@option` values that are supported in a hash (http://www.rubydoc.info/gems/yard/file/docs/Tags.md#option). Those seem worthwhile to consider listing out too.

Reply:
Cool, will fill out the options for the style hash.

Reply:
Hmm, a couple of problems on this one.  Firstly the params aren't showing up in the yard doc because it wants them to be next to an initialize method.

Additionally, the options hash needs to be named in order for it to pick up and match against it.

```
class Blurgh
    # @param thing [Integer] a thing!
    # @param thing_hash [Hash] a thing hash which we want to have.
    # @option thing_hash [String] :blurgh the name of blurgh
    def initialize(thing, thing_hash)
    end
  end
```

I will have to dig deeper into yard to see if I can figure this out...

Reply:
Ah, didn't know that about it. I wouldn't worry on it for too long--if you can't find an easy way, we can table the options hashes until later (or never) and just roll forward with this now.